### PR TITLE
Database labels

### PR DESCRIPTION
**Pull request by q4wen**
Landing page now can accept query token in the form of `<database name>: <id>`

http://appsbeta.pathwaycommons.org/search?gt=0&lt=250&q=hgnc:ATM&type=Pathway and
http://appsbeta.pathwaycommons.org/search?gt=0&lt=250&q=ATM&type=Pathway will have same result.

Ref: #702

### CODE REVIEW

File: src/client/features/search/landing-box.js [hunk 1]
```diff
@@ -122,37 +124,59 @@ synonyms:"TP53, BCC7, LFS1, P53, TRP53"
...
 const getLandingResult= (query)=> {
-  const q=query.trim();
+  let q=query.trim().split(' ');
+  let ncbiIds={},uniprotIds={},labeledId={};
+  const genesToSearch = [];
+  q.forEach ((id)=> {
+    const splitId=id.split(':');
+
+    //Parse the query based on the format
+    if(/uniprot:\w+$/.test(id)) {
```
Comment (jvwong) [inline src/client/features/search/landing-box.js]: Make case insensitive

File: src/client/services/server-api/index.js [hunk 2]
```diff
@@ -25,19 +25,25 @@ const ServerAPI = {
...
-    return fetch('/api/validation', {
-      method:'POST',
-      headers: {
-        'Accept': 'application/json',
-        'Content-Type': 'application/json'
-      },
-      body:JSON.stringify(query)
-    }).then(res => res.json()).then(ids=> _.assign(ids,{unrecognized:_.tail(ids.unrecognized)}));//remove padding
+    if(query.genes.length>=1){
+      query.genes= query.genes;
```
Comment (jvwong) [inline src/client/services/server-api/index.js]: Necessary?

Comment (IgorRodchenkov) [reply src/client/services/server-api/index.js]: Jeff, what do you mean by 'necessary'?

File: src/client/features/search/landing-box.js (hunk 1 above)
Comment (IgorRodchenkov) [reply src/client/features/search/landing-box.js]: i.e. the prefix part (id is case-sensitive)

**Comment by IgorRodchenkov** [comment]
Please, do not insert word "approved" into "HGNC Symbol" (official primary name for that resource, ID type) anywhere.

File: src/client/services/server-api/index.js (hunk 2 above)
Comment (q4wen) [reply src/client/services/server-api/index.js]: "Removing padding" is related to a previous issue(#767) and has been done in pull request(#782). Should I keep the 'padding' and merge the code later?

Comment (jvwong) [reply src/client/services/server-api/index.js]: I'm referring to line 35 `query.genes= query.genes;`. Isn't this assigning a variable reference to itself?

**Comment by IgorRodchenkov** [comment]
Ah, Jeff was quick to merge this before me (I was just testing...) Great!

**Pull request merged**